doc/user-guide/MongooseIM-High-level-Architecture.md

```diff
@@ -0,0 +1,59 @@
+## Inside MongooseIM
+
+MongooseIM comes with a large set of modules to configure and enable. See 'Extension Modules' for more info.
+
+MongooseIM manages two sets of data: transient for session data management, and persistent for archive and configurations.
+
+## MongooseIM and its ecosystem in a datacenter
+
+### Databases
+
+Please refer to 'Database Backends' doc for more configuration information.
+
+* Transient databases
+
+Mnesia will run on the same nodes as MongooseIM.
+
+Redis will be a separate cluster, not using the same nodes as MongooseIM.
+
+* Persistant databases
+
+Both RDBMS (SQL) and NOSQL (Riak KV, Cassandra) 
+
+* LDAP
+
+LDAP will also run on a separate cluster.
+
+### Frontend
+
+Native clients, on platforms such as Android, iOS, Windows, Linux, macOS, will preferrably use a plain XMPP over TCP connections. Web clients cannot use these TCP connections.
```

Comment:
The sentence `Web clients cannot use these TCP connections.` is probably redundant. Web clients are described in the next paragraph.